### ISSUE TITLE
Better Control of Ref Counter in `force_set_current_code`

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/polkadot#3453]
We are about to use `force_set_current_code` to recover the Moonriver parachain, however, we noticed when walking through the code that two reference counters will now be placed on this wasm, whereas it should only have one.

We need to investigate the right patch so this doesnt happen again. Potentially we omit the ref counter addition, or even make it configurable in the extrinsic.

cc @thiolliere

**Comment by rphmeier** [comment]
Furthermore it needs to set up code pruning for the previous code. Or we should add an unsigned extrinsic for cleaning up code without ref counts

**Comment by rphmeier** [comment]
So, to be clear, what happened with the ref-counter is that we had the parachain re-upgrade to the new code C after forcing the code change.

1. At block N the parachain scheduled a code upgrade to C to be applied at block N+K. This set the ref-count of C to 1.
1. At block N+K-1 the parachain upgraded prematurely to C and stalled as a result.
1. The governance of the chain invoked `force_set_current_code` at block X which unconditionally incremented the ref-count of C. This set the ref-count of C to 2.
1. The next block triggered the runtime upgrade to C (again), which is a no-op - except for the fact that `Paras::note_past_code` triggered a past-code-pruning of C.


At the time of writing, there are 2 past-code-pruning events for Moonriver on Kusama.

The first will be triggered at block 8300361 - this is 28,800 blocks after the force-set-current-code and means a pruning of the old code.

The second will be triggered 2 blocks later, at block 8300363 - this is 28,800 blocks after the runtime upgrade was applied for the second time and means that the ref-count of C will be decreased from 2 to 1.

In conclusion: Nothing needs to change in `force_set_current_code` for this specific issue, and the code C will be naturally deleted from the Kusama relay chain when Moonriver upgrades from C to a new runtime C'.

**Comment by shawntabrizi** [comment]
cool, thanks for looking deeper here. Seems the ref counter here is all good then ^.^
